CONTRIBUTING.md

```diff
@@ -1,13 +1,96 @@
-# Pull Request Guidelines
+# Project Guidelines
```

Comment:
Some of these are not practical for small / medium sized teams. A few others are not practical for a scientific software.

Reply:
@ketancmaheshwari Can you please be more specific? Which recommendation you think is not practical and why?